### DISCUSSION TITLE
Additional Data Available?

### DISCUSSION THREAD

**Discussion by arturosLatino** [davetapley/helium-tax#51] [General]
As others have said, thanks Dave for making this, it's very helpful!  I have a question regarding any other data that may be available that you could also include in your report.  Specifically, is there any way to output "expense" type data, like DCs burned and the associated HNT reduction in your wallet?  I recently re-asserted my hotspot,  and "paid" the 1,055,000 DCs for this (1,000,000 DC re-assert fee + 55,000 transaction fee).  I now see a reduction in my wallet of around .28 HNT (using my own math to get this).   I assumed I could find somewhere in the Helium app the actual HNT burned, but so far no luck.  Now when I run your tool, the totals are off by this amount from what's actually in my wallet.  Not too big a deal at the moment, but I imagine over time with many transactions this could be an issue when reporting taxes and deducting these fees.  Thanks for any help!

**Comment by davetapley** [comment]
If you look the wallet up at https://explorer.helium.com/ you should be able to find the transactions, you can filter by type.

As for this tool, I'm not sure if I want to start including 'other stuff'; to my knowledge the only thing required for US taxes is knowing how much income was generated?

**Reply by arturosLatino** [reply]
Hmm, ok.  So far I can find the re-assert action under "Hotspot Transactions", but it simply lists the 1,055,000 DCs burned, not the conversion from DCs to HNT that had to occur for the 'implicit burn'.   

I've written a Matlab GUI that plots my wallet HNT balance (via your tool's output) including each HNT transaction, the USD equivalent, Helium price at the time and totals and averages for the day/week/month/year.  See screenshot attached, hopefully it comes through.  My next endeavor was to include any deductions for things like transfer fees, conversion to usd, etc.   Then my plan was to use this data to figure out all tax reporting. Hopefully I can throw this out on Mathworks if it amounts to anything useful.  

For tax reporting, I'm new to crypto so I'm not totally sure how to do it, but from a brief conversation I had with an accountant last year I think this is what's happening:   There's an IRS defined "correct" way to file crypto taxes, which includes a) keeping track of each helium income and it's dollar equivalent you receive and counting this as ordinary income, then b) keeping track of each USD conversion you make and counting the difference in payment amount to sale amount as a capital gain/loss.   This can get really ugly as you would have to keep track of each incoming HNT increment to know how long you had it before converting it to usd, as capital gains taxes are based on length of time the security (HNT in this case) is owned.  This is the same method used for taxes on stocks.   And this is where a Matlab tool would work great, keeping track of all these individual payments and conversions.  I *think* the accountant mentioned that the IRS was kind of allowing people to just list all the HNT income as either ordinary income at the time of conversion to usd, or maybe just capital gains, due to the complexity of figuring out all the details.  ...But don't quote me on that.  So in brief, I really hope I can discover where to find all associated fees for all transactions in HNT so this tool is accurate. 

![MatTool](https://user-images.githubusercontent.com/96957780/147890964-44fe88eb-e923-48cb-ba52-f319d604e5ce.JPG)

**Reply by davetapley** [reply]
That's a neat looking tool, very cool!

R.e:
> a) keeping track of each helium income and it's dollar equivalent you receive and counting this as ordinary income

Yep, that's what I built this tool for, and is consistent with my understanding too.

And r.e:

> b) keeping track of each USD conversion you make and counting the difference in payment amount to sale amount as a capital gain/loss.

This is my understanding too.
So far I've personally only had gains from selling on exchanges (and so tracked the gains for taxes there),
but (I _presume_) you are correct that a DC burn for e.g. an assert would also count.

The math here is at least simple,
because [One Data Credit will always cost $0.00001](https://docs.helium.com/blockchain/helium-token/#data-credits-and-burn-and-mint-economics),
but I don't know what the tax implications are 😑

**Reply by arturosLatino** [reply]
That's true, I can just deduct $10.55 for the reassert fees on my taxes I guess.

Since I've never transferred or sold any Helium on an exchange, I have a question on your statement "...and so tracked the gains for taxes there":  How does an exchange know the cost basis for the shares that you transfer to the exchange (the amount you were paid originally)? ...wouldn't the exchange only know the value at the time you transferred the tokens over?  
This whole tax thing seems really complex to me, but maybe I'm over thinking it...  Thanks for any insight.

**Reply by davetapley** [reply]
> How does an exchange know the cost basis for the shares that you transfer to the exchange (the amount you were paid originally)?

They don't, and in some sense they _can't_, because there are different ways to decide the basis, [this article has some good info](https://cryptotrader.tax/blog/cryptocurrency-tax-calculations-fifo-and-lifo-costing-methods-explained).

I'd also recommend looking at https://www.minertax.com/, it's not free, but might be worth it for you.

---

> This whole tax thing seems really complex to me

You're not alone, I think part of the problem is that it's relatively easy to acquire crypto, compared to managing it for tax purposes.

Depending on your situation I would consider looking for a CPA.

**Comment by arturosLatino** [comment]
That was a good article, thanks.  I think I have everything I need now... at least to update this tool with everything needed for taxes and plotting.   Your comment about the Data Credit always being $0.00001 dollar was probably the key. 
Thanks a bunch for taking the time to answer my questions!
